sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImplTest.java

```diff
@@ -1091,6 +1092,52 @@ public void testFailInsertOtherRetry() throws Exception {
     verifyWriteMetricWasSet("project", "dataset", "table", "actually forbidden", 1);
   }
 
+  /** Tests that {@link DatasetServiceImpl#insertAll} logs suggested remedy for insert timeouts. */
+  @Test
+  public void testInsertTimeoutLog() throws Exception {
+    TableReference ref =
+        new TableReference().setProjectId("project").setDatasetId("dataset").setTableId("table");
+    List<FailsafeValueInSingleWindow<TableRow, TableRow>> rows = new ArrayList<>();
+    rows.add(wrapValue(new TableRow()));
+
+    setupMockResponses(
+        response -> {
+          when(response.getStatusCode()).thenReturn(400);
+          when(response.getContentType()).thenReturn(Json.MEDIA_TYPE);
+          when(response.getContent())
+              .thenReturn(
+                  toStream(errorWithReasonAndStatus("No rows present in the request.", 400)));
```

Comment:
Nit: let's pad this string in the front and back to better check "e.getMessage().contains".

Reply:
I've added the padding here